sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -81,6 +81,32 @@ private[execution] sealed trait HashedRelation extends KnownSizeEstimation {
    */
   def asReadOnlyCopy(): HashedRelation
 
+
+  /**
+   * Normally HashedRelation is built from an Source (input: Iterator[InternalRow]).
+   * This indicates the origin input is empty.
+   * Note that, the hashed relation can be empty despite the input being not empty,
+   * since the hashed relation skips over null keys.
+   */
+  var isOriginInputEmpty: Boolean
+  def setOriginInputEmtpy(isOriginInputEmpty: Boolean): HashedRelation = {
+    this.isOriginInputEmpty = isOriginInputEmpty
+    this
+  }
+
+  /**
+   * It's only used in null aware anti join.
+   * This will be set true if Source (input: Iterator[InternalRow]) contains a key,
+   * which is allNullColumn.
+   */
+  var allNullColumnKeyExistsInOriginInput: Boolean
+  def setAllNullColumnKeyExistsInOriginInput(
+      allNullColumnKeyExistsInOriginInput: Boolean): HashedRelation = {
+    this.allNullColumnKeyExistsInOriginInput =
```

Comment:
nit: Should this line be split into two ?

Reply:
done